### DISCUSSION TITLE
ws2812(array) command and RGBW strip

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#13184] [Q&A] [answered]
Question about Scripting and the ws2812(array) command; I have an rgbw strip, so using 0x00110011 is using 8 instead of the usual 6 numbers, however I am not able to control my strip with the white led's using the ws2812(array) command. When filling and array with 0xff0000 it works, but filling it for example with 0x000000ff does not turn on the white led's?

The strip works fine with all other functions, from the UI, Led<x>, Color etc commands work fine with all colors

**Comment by gemu2015** [comment, marked as answer]
the standard ws2812 only supports RGB which is 24 bits. scripting uses float for all numbers and arrays. float can only hold 24 bits of integer. so RGBW 32 bits is not supported.
to support RGBW 2 floats of the arrays would be needed (but wastes some memory). this is doable but currently not implemented.

**Comment by martin072** [comment]
Thank you for the explanation. Would there be a way to choose which 24 bits I would like to use? i.e lets say I don't want the Green color, but want to use the red blue and white only, so not RGB but RWB (or RBW)? would that be possible?

**Comment by gemu2015** [comment]
that would be more complicated than 2 array values per pixel.

i modified the command to specify 32 bits with 2 consecutive array values.
1 value = 16 bits R+G
2 value = 16 bits B+W

select with ws2812(array 0x1000)

keep in mind that the array size must be twice as large as the number of pixels now
you may enlarge the allowed array size with
#define LARGE_ARRAYS

try if it works for you

[xdrv_10_scripter.ino.zip](https://github.com/arendst/Tasmota/files/7194855/xdrv_10_scripter.ino.zip)

**Reply by martin072** [reply]
Do I need to download the development release for this ? I am currently running 9.4.0 on the device with the led bar

**Reply by martin072** [reply]
Made this little test script, does not do anything when supplying the 0x1000 parameter. Without it works fine.

```
>D
;90 LEDs so double array
m:array=0 180
cnt=0
col1A=0xff00
col1B=0x00ff
col2A=0x0000
col2B=0x00ff

>S
for cnt 0 180 2
array[cnt]=col1A
next

for cnt 1 179 2
array[cnt]=col1B
next

ws2812(array 0x1000)
delay(500)

for cnt 0 180 2
array[cnt]=col2A
next

for cnt 1 179 2
array[cnt]=col2B
next

ws2812(array 0x1000)
delay(500)

```

**Comment by gemu2015** [comment]
sorry, use this version

[xdrv_10_scripter.ino.zip](https://github.com/arendst/Tasmota/files/7195201/xdrv_10_scripter.ino.zip)

**Reply by martin072** [reply]
Does work now :) , but the order is different
1st array is BW
2nd array is RG

**Comment by martin072** [comment]
@gemu2015 

will his be included in the next tasmota versions? Or would it require me to stick on this specific one?

thnaks

**Reply by gemu2015** [reply]
already merged into development branch
